src/server/pfs/server/testing/server_test.go

```diff
@@ -1178,6 +1178,50 @@ func TestPFS(suite *testing.T) {
 		require.True(t, strings.Contains(err.Error(), "branch and head commit must belong to the same repo"))
 	})
 
+	suite.Run("DeleteProject", func(t *testing.T) {
+		t.Parallel()
+		ctx := pctx.TestContext(t)
+		env := realenv.NewRealEnv(ctx, t, dockertestenv.NewTestDBConfig(t))
+
+		_, err := env.PachClient.PfsAPIClient.CreateProject(ctx, &pfs.CreateProjectRequest{Project: &pfs.Project{Name: "test"}})
+		require.NoError(t, err)
+		_, err = env.PachClient.PfsAPIClient.DeleteProject(ctx, &pfs.DeleteProjectRequest{Project: &pfs.Project{Name: "test"}})
+		require.NoError(t, err)
+	})
+
+	suite.Run("DeleteProjectWithRepos", func(t *testing.T) {
+		t.Parallel()
+		ctx := pctx.TestContext(t)
+		env := realenv.NewRealEnv(ctx, t, dockertestenv.NewTestDBConfig(t))
+
+		_, err := env.PachClient.PfsAPIClient.CreateProject(ctx, &pfs.CreateProjectRequest{Project: &pfs.Project{Name: "test"}})
+		require.NoError(t, err)
+		_, err = env.PachClient.PfsAPIClient.CreateRepo(ctx, &pfs.CreateRepoRequest{Repo: &pfs.Repo{Project: &pfs.Project{Name: "default"}, Name: "test"}})
+		require.NoError(t, err)
+		_, err = env.PachClient.PfsAPIClient.CreateRepo(ctx, &pfs.CreateRepoRequest{Repo: &pfs.Repo{Project: &pfs.Project{Name: "test"}, Name: "test"}})
+		require.NoError(t, err)
+		_, err = env.PachClient.PfsAPIClient.DeleteProject(ctx, &pfs.DeleteProjectRequest{Project: &pfs.Project{Name: "test"}})
+		require.YesError(t, err)
```

Comment:
Maybe also verify the error message with something like 
```Go
require.ErrorContains(t, err, "project test contains 2 repos")
```

Reply:
I think that I’d prefer not to hardcode that at the moment. Longer term I think that we need to attack structured errors across RPCs, but … that’s another story.